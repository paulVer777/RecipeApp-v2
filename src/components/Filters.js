import React from 'react'
import { connect } from 'react-redux';
import { changeFilters } from '../actions/filterAction';

class Filters extends React.Component {
    
    state = {
        text:'',
        difficulty:'all',
        sortBy:'alphabetical'
    }

    onEventHandler = (event) => {
        const data = event.target.value
        const propertyName = event.target.name
        this.setState((prevState) => ({[propertyName]:data}), ()=> this.props.changeFilters(this.state))
    }

    render(){
        return (
            <div className = 'filters' >
                <input type = 'text' placeholder = 'Search (recipe name)' name = 'text' onChange = {this.onEventHandler} />
                <select name = 'difficulty' onChange = {this.onEventHandler}>
                    <option value = 'all' >All</option>
                    <option value = 'easy' >Easy</option>
                    <option value = 'medium'>Medium</option>
                    <option value = 'hard'>Hard</option>
                </select>
                <select name = 'sortBy' value = {this.state.sortBy} onChange = {this.onEventHandler} >
                    <option value = 'alphabetical'>Alphabetical</option>
                    <option value = 'preparationTime'>Preparation Time</option>
                </select>
            </div>
        )
    }
}


const mapDispatchToProps = (dispatch) => (
    {
        changeFilters:(data) => dispatch(changeFilters(data))
    }
)




export default connect(undefined,mapDispatchToProps)(Filters)