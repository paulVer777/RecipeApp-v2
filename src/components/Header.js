import React from 'react'
import {Link} from 'react-router-dom'

const Header = () => (
    <div className = 'header'>
        <Link to = {'/'} >Home</Link>
        <Link to = {'/my-ingredients'} >My Ingredients</Link>
    </div>
)

export default Header