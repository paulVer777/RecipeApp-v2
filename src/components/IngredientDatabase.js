import React from 'react'

const IngredientDatabase = () => (
    
    <div>
        MyIngredients
    </div>
)
    
export default IngredientDatabase