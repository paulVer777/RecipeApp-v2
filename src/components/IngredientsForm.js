import React from 'react'
import { connect } from 'react-redux';
import { addIngredient } from '../actions/recipeActions';
import uuidv4 from 'uuid'

 export class IngredientsForm extends React.Component {

state = {
    text: ''
}

onSubmitHandler = (event) => {
        event.preventDefault()
        const data = event.target.elements.text.value
        this.setState((prevState) => ({text:data}),() => this.props.addIngredient(this.props.id,{title:data, id:2}))
        
}

render(){
    return(
        <div className = 'ingredient-form' >
        <form onSubmit = {this.onSubmitHandler}>
            <input type = 'text' placeholder = 'ingredient name' name = 'text' />
            <button>Add Ingredient</button>
        </form>
        </div>
    )
}
}


const mapDispatchToProps = (dispatch) => ({
    addIngredient:(id,ingredientObj)=> dispatch(addIngredient(id,ingredientObj))
})

export default connect(undefined,mapDispatchToProps)(IngredientsForm)
 
