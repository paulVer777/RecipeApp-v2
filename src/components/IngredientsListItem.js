import React from 'react'
import {connect} from 'react-redux'
import { removeIngredient } from '../actions/recipeActions';

const IngredientsListItem = (props) => (
    <div className = 'ingredient-list-item'>
        <span>{props.data.title}</span>
        <button onClick = { () => props.dispatch(removeIngredient(props.recipeId,props.data.id))}>Remove</button>
    </div>
)

export default connect()(IngredientsListItem)