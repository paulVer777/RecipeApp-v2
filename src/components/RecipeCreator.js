import React from 'react'
import uuidv4 from 'uuid'
import {connect} from 'react-redux'
import {addRecipe} from '../actions/recipeActions'
import {Link} from 'react-router-dom'

const RecipeCreator = (props) => {

    const id = uuidv4()

    return (
    <div className = 'recipe-creator'>
        <h3>Recipe creator</h3>
        <Link 
        className = 'recipe-creator__link'
        to = {`/recipe-editor/${id}`}
        onClick = {() => {props.dispatch(addRecipe(id))}}
        >Make somee goooddss
        </Link>
    </div>
    )
}

export default connect()(RecipeCreator)