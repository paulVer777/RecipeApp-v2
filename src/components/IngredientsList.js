import React from 'react'
import { connect } from 'react-redux';
import IngredientsListItem from '../components/IngredientsListItem'

const IngredientsList = (props) => (
    <div className = 'ingredient-list' >
        <h3>Ingredients</h3>
        { props.recipe.ingredients.length < 1 ? 'You have no ingredients added yet' 
            : 
        props.recipe.ingredients.map((value,index) => <IngredientsListItem  data = {value} key = {value.id} />)}        
    </div>
)

const mapStateToprops = (state,props) => (
    {
        recipe:state.recipes.find((value, index) => props.id === value.id)
    }
)

export default connect(mapStateToprops)(IngredientsList)