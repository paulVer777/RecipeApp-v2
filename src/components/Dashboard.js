import React from 'react'
import RecipeCreator from '../components/RecipeCreator'

const Dashboard = () => (
    <div className = 'dashboard'>
        <h2>Dasboard</h2>
        <RecipeCreator />
    </div>
)
    
export default Dashboard