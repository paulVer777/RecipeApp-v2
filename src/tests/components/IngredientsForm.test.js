import React from 'react'
import  {shallow} from 'enzyme'
import {IngredientsForm} from '../../components/IngredientsForm';


test('should render IngredientsForm correctly' , () => {
    const wrapper = shallow(<IngredientsForm />)
    expect(wrapper).toMatchSnapshot()
})

test('Should call add ingredient with correct data', ()=>{

    const addIngredient = jest.fn()

    const wrapper =  shallow( <IngredientsForm addIngredient = {addIngredient} id = {2} /> )
    wrapper.find('form').simulate('submit', {
        preventDefault:()=> {},
        target:{
            elements:{
                text:{
                    value:'hej'
                }
            }
        }
    })
    expect(addIngredient).toHaveBeenLastCalledWith(2,{title:'hej',id:2})
})