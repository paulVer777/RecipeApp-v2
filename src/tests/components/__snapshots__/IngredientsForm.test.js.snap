// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render IngredientsForm correctly 1`] = `
<div
  className="ingredient-form"
>
  <form
    onSubmit={[Function]}
  >
    <input
      name="text"
      placeholder="ingredient name"
      type="text"
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;
